pkg/smi/fake.go

```diff
@@ -59,13 +58,13 @@ func (f fakeMeshSpec) ListServiceAccounts() []service.K8sServiceAccount {
 }
 
 // GetService fetches a specific service declared in SMI for the fake Mesh Spec.
-func (f fakeMeshSpec) GetService(svc service.MeshService) (service *corev1.Service, err error) {
+func (f fakeMeshSpec) GetService(svc service.MeshService) *corev1.Service {
```

Comment:
It seems that here we switch into implicit mode - where by convention if the reference is nil, the service does not exist (no explicit error) - and perhaps that's totally acceptable.

Reply:
I thought about this, and concluded that we have a precedent for this where we return empty lists when we encounter errors. 
Ex.
```
func SomeFunc() []someList {
    err := doSomething()
    if err != nil {
        log.Error()...
        return nil
    }
    return nonEmptyList
}
```

I guess for nil return types this is a reasonable assumption for the caller to make.